### DISCUSSION TITLE
An easy way to look up old documentation?

### DISCUSSION THREAD

**Discussion by ByteOtter** [pypa/setuptools#3864] [Q&A]
Hello everyone!

I hope this is the right place to ask this:
I am currently working on something for [Cobbler](https://github.com/cobbler) and as we still use Python 3.6 for reasons beyond our control I needed to refer to the documentation of Setuptools 44. So I went to your official documentation and hoped to find the old docs there. Sadly though, I did not.
Eventually I went to the way back machine and pulled up [the docs from January 2020](https://web.archive.org/web/20200118161529/https://setuptools.readthedocs.io/en/latest/setuptools.html#using-a-src-layout) which seems to be exactly what I needed.

I just wondered if there was an easier way to find this stuff that I just missed?
If not I think it would be nice if you could archive this somewhere as a lot of people still have to rely on older Python versions for their customers. 🤔
Let me know what you think. ✌🏼

**Comment by abravalheri** [comment]
Hi @ByteOtter, I am afraid that keeping a long term cache of the published documentation for all versions is incompatible with the [setuptools release process](https://setuptools.pypa.io/en/latest/development/releases.html).

If you want to make sure you are reading the documentation exactly how it was published by the time the release was uploaded to PyPI, I think your best bet would be cloning the specific tag of setuptools (or downloading the relevant `sdist`) and re-building the docs (usually `tox -e docs`, except for very old versions).

If you don't mind reading unparsed rST files, you can also try having a look on [inspector](https://inspector.pypi.io/project/setuptools/44.1.1/packages/b2/40/4e00501c204b457f10fe410da0c97537214b2265247bc9a5bc6edd55b9e4/setuptools-44.1.1.zip/setuptools-44.1.1/docs/setuptools.txt).

**Reply by ByteOtter** [reply]
thank you for your swift response.
As far as I know Readthedocs can build the documentation for each tag for you. So would this be a huge effort to keep at least the last two major version's docs?
Sorry if this is pushing it I just want to get my head around this.

**Reply by abravalheri** [reply]
Hi @ByteOtter do you know what if RTD has a configuration for "keep the last 2 major versions"?

**Reply by SchoolGuy** [reply]
Yes, take a look at their docs [here](https://docs.readthedocs.io/en/stable/versions.html#hidden). They describe this method as "Hidden" Docs. They also were as nice as having a dedicated small tutorial for that purpose: [Hiding a version](https://docs.readthedocs.io/en/stable/guides/hiding-a-version.html).

I am aware that they don't let you control that via an API/configuration but I am sure that your project is big enough that if you kindly ask them they will at least consider such a feature request.

**Reply by abravalheri** [reply]
Hi @SchoolGuy, thank you very much for the links.

This is not exactly what I had in mind in my question 😢. I was wondering if there was a checkbox in RTD that said "keep the last 2 major versions and discard the other ones".

Based on the documentation that you shared, it seems that implementing something like that would require constant management actions upon the docs (please correct me if I had a wrong impression). Even if there was a API for a programmatic approach, it would require us to implement and maintain a CI system for that (I would prefer spending this "effort" working on the other issues/plans for setuptools).

To be honest, I think that the gains of such approach would be very marginal. Having a look on the [history](https://setuptools.pypa.io/en/latest/history.html) of the project, we can see that the latest 2 major versions are about 2 months away from each other and they don't really differ significantly in terms of documentation...

My personal opinion is that, given setuptools' chosen [release process](https://setuptools.pypa.io/en/latest/development/releases.html) and its implications, the suggested approach would not bring enough benefits to go to the RTD team with a feature request...

Of course if any member of the community would like to champion this with RTD, this might be more useful in contexts other than setuptools.

**Reply by SchoolGuy** [reply]
I believe it is very useful, even if no huge changes appear. It enables you to look at exactly the versions of documentation that you currently can use. Especially in the corporate environment where you cannot just upgrade to a new major version because you want to, this will make things a lot easier. I will see what @ByteOtter and I can move. Maybe the RTD people have an idea of what to do with such a wish.
